src/Workspaces/Core/Portable/Workspace/Host/PersistentStorage/IPersistentStorageConfiguration.cs

```diff
@@ -52,12 +54,11 @@ protected virtual string GetCacheDirectory()
             return Path.Combine(appDataFolder, "Microsoft", "VisualStudio", "Roslyn", "Cache");
         }
 
-        public string? TryGetStorageLocation(Solution solution)
-            => TryGetStorageLocation(solution.Workspace, SolutionKey.ToSolutionKey(solution));
+        public bool ThrowOnFailure => false;
```

Comment:
I'm all good with this, though this would be nice to doc.  However, i'm happy to do that doc myself.

Reply:
Will add a comment in follow up PR.

Reply:
Actually, the comment is on the interface method.

Reply:
sure.  though in this case i would like the comment to be more about why we're 'false' here.  it's clear to me because i know the hsitory.  but i imagine someone going: wait... by default, are we not throwing in failures, taht seems bad.

Reply:
The comment on the interface explains that though:

> Indicates that the client expects the DB to succeed at all work and that it should not ever gracefully fall over. Should not be set in normal host environments, where it is completely reasonable for things to fail (for example, if a client asks for a key that hasn't been stored yet).

Reply:
Perhaps the negative - `RecoverFromFailure` would be more self-descriptive?

Reply:
fair :)